### PR TITLE
feat(virtual-scroll): add virtual-scroll component

### PR DESCRIPTION
**Pull request by necoxrr**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Added a new virtual scroll component for Vue applications to enhance performance with large datasets.
  - Introduced task configuration for TypeScript builds in Visual Studio Code.
  
- **Updates**
  - Integrated `VirtualScroll` and `VirtualScrollPc` in modules to improve desktop interface functionality.
  - Updated dependencies to include `@opentiny/vue-virtual-scroll`.

- **Enhancements**
  - Improved component import and export structures for better modularity and maintainability.
  - Enhanced style management with new LESS files for virtual scrolling.

### CODE REVIEW

**Review by ErKeLost — APPROVED** [review]

**Comment by kagol** [comment]
@necoxrr 虚拟滚动是一个新组件，先参考[这个文档](增加新组件的步骤)，走通组件流程，确保执行 `pnpm site` 命令之后，打开的本地网站左侧组件导航能看到这个组件，并且可以看到组件的demo/api文档。

请参考：[增加新组件的步骤](https://github.com/opentiny/tiny-vue/wiki/%E5%A2%9E%E5%8A%A0%E6%96%B0%E7%BB%84%E4%BB%B6%E7%9A%84%E6%AD%A5%E9%AA%A4)

File: packages/renderless/src/virtual-scroll/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+// 具体操作方法和工具函数
+export const initUser =
```
Comment (kagol) [inline packages/renderless/src/virtual-scroll/index.ts]: @necoxrr 这个函数跟虚拟滚动似乎无关，包括以下的 `showCard` 函数

File: packages/renderless/src/virtual-scroll/vue.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { initUser, showCard, showDetail, computedTextField, computedValueField } from './index'
+// 定义该组件暴漏出的接口方法
+export const api = ['state', 'showCard', 'showDetail']
+// 管理virtual-scroll相关的逻辑和状态，不涉及实际渲染部分
+export const renderless = (props, { reactive, watch, computed }, { service }) => {
+  service = initService({ props, service })
+
+  const api = {}
+// 初始化状态
+  const state = reactive({
```
Comment (kagol) [inline packages/renderless/src/virtual-scroll/vue.ts:33]: @necoxrr state是空的，模板里用到的变量和方法都没有声明，这个PR没有跑通虚拟滚动组件最基本的流程

File: packages/renderless/src/virtual-scroll/index.ts (hunk 1 above)
Comment (necoxrr) [reply packages/renderless/src/virtual-scroll/index.ts]: 已经删除无关函数，增添了虚拟滚动相关逻辑函数

File: packages/renderless/src/virtual-scroll/vue.ts (hunk 2 above)
Comment (necoxrr) [reply packages/renderless/src/virtual-scroll/vue.ts:33]: 已解决

**Review by kagol — APPROVED** [review]

File: packages/renderless/src/virtual-scroll/index.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ * Use of state source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+// start, end
+// 二分法查找
+const binarySearch = (list, value) => {
```
Comment (ErKeLost) [inline packages/renderless/src/virtual-scroll/index.ts:14]: 类似这种比较通用的方法建议放到 utils 里 比如可以创建一个 utils 文件 与 组件功能无关的都有可以放进去

Comment (ErKeLost) [inline packages/renderless/src/virtual-scroll/index.ts:14]: 建议这些方法都带上类型，因为涉及组件属性传递等方面，也方便后续维护是更通俗易懂

File: packages/renderless/src/virtual-scroll/index.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    console.log(dValue, nodes.length, positions.length, 'd')
+    if (!dValue || !nodes.length) return
+    for (let k = nodes.length; k < positions.length; k++) {
+      positions[k].top = positions[k - 1].bottom
+      positions[k].bottom = positions[k].bottom - dValue
+    }
+  }
+
+// 计算数据总高度
+export const calculateTotalSize =
```
Comment (ErKeLost) [inline packages/renderless/src/virtual-scroll/index.ts:76]: 这段代码可以优化成 

```ts
export const calculateTotalSize = ({ props, state }) => () => {
  if (props.itemSize) {
    return props.itemSize * state.data.length;
  }
  
  const lastPosition = state.positions?.[state.positions.length - 1];
  return lastPosition ? lastPosition.bottom : 0;
};
```

File: packages/renderless/src/virtual-scroll/index.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+// 虚拟滚动逻辑
+export const handleScroll = ({ props, state, virtualScroll, nextTick, items, ...rest }) => {
+  return async (event) => {
+    if (!virtualScroll.value) return
+    await nextTick()
+    const size = virtualScroll.value.getBoundingClientRect()[props.direction === 'vertical' ? 'height' : 'width']
+    const viewNum = Math.floor(size / (props.itemSize ?? props.estimatedItemSize))
+    // console.log(virtualScroll.value.getBoundingClientRect().height, viewNum, state)
```
Comment (ErKeLost) [inline packages/renderless/src/virtual-scroll/index.ts:95]: 无用代码可以删除

**Pull request merged**